### PR TITLE
Path Traversal via Symlink in @hapi/hapi

### PR DESCRIPTION
**Pull request by vineetpandey**
## ✍️ Description

Path Traversal via Symlink in @hapi/hapi

## 🕵️‍♂️ Proof of Concept

Steps To Reproduce:
Install the hapi: npm i @hapi/hapi

Create a symlink file within the directory
ln -s /etc/shadow test_shadow

3. Request the file within browser
http://localhost:3000/test_shadow

## 💥 Impact

It allows attacker to read content of arbitrary file on remote server and could leverage attacks like remote code execution.

## ☎️ Contact

No, didn't reached out to anyone yet. 

## ✅ Checklist

**In my pull request, I have:**

- [x] _Created and populated the `README.md` and `vulnerability.json` files_
- [x] _Provided the repository URL and any applicable [permalinks]([https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files](https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files))_
- [x] _Defined all the applicable weaknesses ([CWEs]([https://cwe.mitre.org/](https://cwe.mitre.org/)))_
- [x] _Proposed the CVSS vector items i.e. User Interaction, Attack Complexity_
- [x] _Checked that the vulnerability affects the latest version of the package released_
- [x] _Checked that a fix does not currently exist that remediates this vulnerability_
- [x] _Complied with all applicable laws_

### CODE REVIEW

**Comment by adam-nygate** [comment]
Hi @vineetpandey, thanks again for the disclosure!

Our team are still debating whether this Path Traversal vulnerability is valid. Arguably, a developer who uses `hapi` may have legitimate use cases for why they would want to use a symlink to expose certain files, and so a blanket fix for this being to prevent symlinking would not be useful.

Do you have any thoughts on this? What we think might be helpful is if we engage with the hapijs team to get their opinion on this, and if they agree with the validity of the vulnerability, we can reward you for the disclosure, and whoever fixes it.

**Comment by vineetpandey** [comment]
Thanks @mufeedvh/@toufik-airane for acknowledging the issue.

File: bounties/npm/@hapi/1/vulnerability.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{
+    "PackageVulnerabilityID": "1",
+    "DisclosureDate": "2020-06-05",
+    "AffectedVersionRange": "19.1.1",
```
Comment (adam-nygate) [inline bounties/npm/@hapi/1/vulnerability.json:4]: Can you please confirm the affected range. Generally, we only accept disclosures for vulnerabilities that affect all (`*`) versions and or the latest version.

File: bounties/npm/@hapi/1/vulnerability.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    "Summary": "Path traversal in @hapi/hapi via Symlink",
+    "Author": {
+        "Username": "vineetpandey",
+        "Name": "Vineet Pandey"
+    },
+    "Package": {
+        "Registry": "npm",
+        "Name": "@hapi/hapi",
```
Comment (adam-nygate) [inline bounties/npm/@hapi/1/vulnerability.json:12]: Replace with just `hapi`

File: bounties/npm/@hapi/1/vulnerability.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        "RC": "C",
+        "Score": "7.5"
+    },
+    "CVEs": [
+        ""
+    ],
+    "Repository": {
+        "URL": "https://hapijs.com/",
+        "Codebase": [
+            "https://github.com/hapijs/hapi"
```
Comment (adam-nygate) [inline bounties/npm/@hapi/1/vulnerability.json:40]: Replace with primary codebase language - `JS`?

File: bounties/npm/@hapi/1/vulnerability.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        "E": "F",
+        "RL": "",
+        "RC": "C",
+        "Score": "7.5"
+    },
+    "CVEs": [
+        ""
+    ],
+    "Repository": {
+        "URL": "https://hapijs.com/",
```
Comment (adam-nygate) [inline bounties/npm/@hapi/1/vulnerability.json:38]: Replace with URL to the GitHub repository

File: bounties/npm/@hapi/1/vulnerability.json [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        "Codebase": [
+            "https://github.com/hapijs/hapi"
+        ]
+    },
+    "Permalinks": [
+        "https://github.com/hapijs/hapi"
```
Comment (adam-nygate) [inline bounties/npm/@hapi/1/vulnerability.json:44]: Please use this to provide permalinks for the specific issue

File: bounties/npm/@hapi/1/vulnerability.json (hunk 1 above)
Comment (vineetpandey) [reply bounties/npm/@hapi/1/vulnerability.json:4]: Yes. this is the latest version - 19.1.1

**Comment by vineetpandey** [comment]
@adam-nygate can you please merge these changes. Thanks!

**Comment by adam-nygate** [comment]
> @adam-nygate can you please merge these changes. Thanks!

Hey @vineetpandey, still waiting for you to resolve a few of the issues I mentioned in my review (e.g. https://github.com/418sec/huntr/pull/50#discussion_r444999142)

File: bounties/npm/@hapi/1/vulnerability.json (hunk 4 above)
Comment (vineetpandey) [reply bounties/npm/@hapi/1/vulnerability.json:38]: Updated it.

File: bounties/npm/@hapi/1/vulnerability.json (hunk 2 above)
Comment (vineetpandey) [reply bounties/npm/@hapi/1/vulnerability.json:12]: Updated it.

File: bounties/npm/@hapi/1/vulnerability.json (hunk 3 above)
Comment (vineetpandey) [reply bounties/npm/@hapi/1/vulnerability.json:40]: Updated it.

File: bounties/npm/@hapi/1/vulnerability.json (hunk 5 above)
Comment (vineetpandey) [reply bounties/npm/@hapi/1/vulnerability.json:44]: Updated it.

**Comment by vineetpandey** [comment]
@adam-nygate - did it

File: bounties/npm/@hapi/1/vulnerability.json (hunk 2 above)
Comment (adam-nygate) [reply bounties/npm/@hapi/1/vulnerability.json:12]: The latest commit does not show any updates.

File: bounties/npm/@hapi/1/vulnerability.json (hunk 4 above)
Comment (adam-nygate) [reply bounties/npm/@hapi/1/vulnerability.json:38]: The latest commit does not show any updates.

File: bounties/npm/@hapi/1/vulnerability.json (hunk 3 above)
Comment (adam-nygate) [reply bounties/npm/@hapi/1/vulnerability.json:40]: The latest commit does not show any updates.

File: bounties/npm/@hapi/1/vulnerability.json (hunk 5 above)
Comment (adam-nygate) [reply bounties/npm/@hapi/1/vulnerability.json:44]: The latest commit does not show any updates.

**Comment by adam-nygate** [comment]
> @adam-nygate - did it

Can you check if you've correctly pushed your commit as I cannot see the changes on the PR

**Comment by vineetpandey** [comment]
@adam-nygate, I believe you should check "staging1" branch rather than staging branch. I apologize since on staging branch wasn't let me push the changes when long back I was doing it.

Still if you don't see the changes then I'll create another another issue and close this one. Hope that might help.

**Comment by adam-nygate** [comment]
> @adam-nygate, I believe you should check "staging1" branch rather than staging branch. I apologize since on staging branch wasn't let me push the changes when long back I was doing it.
> 
> Still if you don't see the changes then I'll create another another issue and close this one. Hope that might help.

@vineetpandey I've looked at your repository and it looks like the commit with the changes is your `master` branch, and not `staging`/`staging1`. Can you modify this PR to be based on your `master` branch?

**Comment by vineetpandey** [comment]
@adam-nygate  - I'm closing this one and created another PR based master branch.
https://github.com/418sec/huntr/pull/158

**Pull request closed without merging**